smart-contracts/contracts/Mocks/CommissionToken.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+/**
+ * @title Commission Token
+ * @dev This token will charge a programmable fee on every transfer that is credited to the the devolpoers address
+ *      this test token will help test various tokens that may transfer a different amount then the transfer requests
+ *      to the recipient.
+ **/
+contract CommissionToken is ERC20 {
```

Comment:
We should probably have an integration test for this?

Reply:
We can, a commission token will have the same flows like any other ERC20 token in our codebase so long as the events emitted are the same, which I can extend the commission test to check for. But more testing never hurts.